### DISCUSSION TITLE
Telit FN980 UE

### DISCUSSION THREAD

**Discussion by yc541** [srsran/srsRAN_Project#796] [General Help] [answered]
I have been testing the latest build with a Telit FN980 UE and the UE always loses connection after registration complete message. I have a few other 5G smart phones which are working fine, Oneplus Nord 5G, Xiaomi 12 etc...

Looking at the gnb logs, there are a lot of invalid pucch format 1 PDUs, mostly followed with a valid pucch format 2 PDU, then later the format 2 PDU becomes invalid as well. I have tested with all firmware on telit website (https://www.telit.com/fn980m-download-zone/) and the results are the same. I wonder if there are any pucch configurations can be tweaked to further debug the issue?

The radio is a USRP X310 with external 10MHz and 1 PPS from an Octoclock.

The logs and pcaps are attached. Thank you.

[telitlogs.zip](https://github.com/user-attachments/files/16786282/telitlogs.zip)

**Comment by robertfalkenberg** [comment]
In the MAC PCAP I see the UE did not send any `PDU  session establishment request` after registration:

```
15	15:34:24,120469			NR RRC/NAS-5GS	115			DL Information Transfer, Configuration update command  [55-bytes]  (Padding 24 bytes) 
16	15:34:24,148604			RLC-NR	575			UEId=1     [UL] [AM] SRB:1  [CONTROL]  ACK_SN=4       (Short BSR LCG ID=0 BS=0)   (PHR PH=35 PCMAX_f_c=52)   (Padding 533 bytes) 
```

Instead the UE sends periodic power headroom reports for almost exactly 10s; then silently leaves the cell:

```
17	15:34:25,108641			MAC-NR	575			UL-SCH (Short BSR LCG ID=0 BS=0)   (PHR PH=33 PCMAX_f_c=52)   (Padding 538 bytes) 
18	15:34:25,488717			MAC-NR	575			UL-SCH (Short BSR LCG ID=0 BS=0)   (PHR PH=35 PCMAX_f_c=52)   (Padding 538 bytes) 
19	15:34:27,508640			MAC-NR	575			UL-SCH (Short BSR LCG ID=0 BS=0)   (PHR PH=36 PCMAX_f_c=52)   (Padding 538 bytes) 
20	15:34:30,588647			MAC-NR	575			UL-SCH (Short BSR LCG ID=0 BS=0)   (PHR PH=35 PCMAX_f_c=52)   (Padding 538 bytes) 
21	15:34:31,048630			MAC-NR	575			UL-SCH (Short BSR LCG ID=0 BS=0)   (PHR PH=35 PCMAX_f_c=52)   (Padding 538 bytes) 
22	15:34:31,428646			MAC-NR	575			UL-SCH (Short BSR LCG ID=0 BS=0)   (PHR PH=35 PCMAX_f_c=52)   (Padding 538 bytes) 
23	15:34:33,968645			MAC-NR	575			UL-SCH (Short BSR LCG ID=0 BS=0)   (PHR PH=34 PCMAX_f_c=52)   (Padding 538 bytes) 
// UE stopped sending...
24	15:34:35,995542			MAC-NR	36			DL-SCH (TAG=0 TA=35)   (Padding 2 bytes) 
```

Message 23 is also the last lifesign from the UE:
```
2024-08-28T15:34:33.968513 [PHY     ] [D] [   781.7] PUSCH: rnti=0x4601 h_id=0 prb=[3, 9) symb=[0, 14) mod=64QAM rv=0 tbs=544 crc=OK iter=2.0 sinr=33.4dB t=121.3us uci_t=0.0us ret_t=0.0us
```
The next scheduled UL transmission is KO, which is exactly 10s after message 15:
```
2024-08-28T15:34:34.128576 [PHY     ] [D] [   797.7] PUSCH: rnti=0x4601 h_id=0 prb=[3, 9) symb=[0, 14) mod=64QAM rv=0 tbs=544 crc=KO iter=6.0 sinr=-21.6dB t=176.0us uci_t=0.0us ret_t=0.0us
```

Please check if the UE or the host system of the UE are properly configured to create a PDU session after successful attach, e.g. the APN configuration.

**Reply by yc541** [reply, marked as answer]
Many thanks for the debug. We have the Telit module on a Quectel evaluation board and the UE does not send the request when the board is connected to a Windows machine. When the board is connected to a Linux machine (using Quectel Connection Manager), the Telit module sends the request and attaches successfully. There may be some Windows driver issues but at least it works with Linux.

**Reply by robertfalkenberg** [reply]
Thanks for the feedback towards the community! Well done!
